docs/build/guides/archival/extend-persistent-entry-js.mdx

```diff
@@ -0,0 +1,179 @@
+
+- For a persistent entry, specify the contract address, storage key, and durability type (persistent)
+- For contract code, specify the contract ID
+
+2. Create a footprint and declare which storage locations will be extended and include the identifier for persistent entry, contract code, or both, depending on your intent.
+
+3. Build and submit the transaction:
+
+- Add the `extendFootprintTtl` operation with your intended TTL
+- Sign and submit the transaction
+
+:::note TTL can only be extended by the deployer or an authorized account. :::
+
+### Extending a persistent entry
+
+```javascript
+const StellarSdk = require("@stellar/stellar-sdk");
+const { Server } = require("@stellar/stellar-sdk/rpc");
+
+async function extendPersistentEntryTTL(contractId, storageKey, sourceKeypair) {
+  const server = new Server("https://soroban-testnet.stellar.org");
+  const account = await server.getAccount(sourceKeypair.publicKey());
+  const fee = "200100";
+
+  // Create an identifier for the entry
+  const persistentEntry = new StellarSdk.xdr.LedgerKey.contractData({
+    contract: StellarSdk.Address.fromString(contractId).toScAddress(),
+    key: StellarSdk.xdr.ScVal.scvSymbol(storageKey),
+    durability: StellarSdk.xdr.ContractDataDurability.persistent(),
+  });
+
+  // Create footprint for the entry
+  const footprint = new StellarSdk.xdr.LedgerFootprint({
+    readOnly: [],
+    readWrite: [persistentEntry],
+  });
+
+  // build and submit the transaction
+  const transaction = new StellarSdk.TransactionBuilder(account, {
+    fee,
+    networkPassphrase: StellarSdk.Networks.TESTNET,
+  })
+    .addOperation(
+      StellarSdk.Operation.extendFootprintTtl({
+        extendTo: 500_000,
+        footprint: footprint,
+      }),
+    )
+    .setTimeout(30)
+    .build();
+
+  transaction.sign(sourceKeypair);
+  return await server.sendTransaction(transaction);
+}
+```
+
+### Extending the contract code
+
+```javascript
+const StellarSdk = require("@stellar/stellar-sdk");
+const { Server } = require("@stellar/stellar-sdk/rpc");
+
+async function extendContractCodeTTL(contractId, sourceKeypair) {
+  const server = new Server("https://soroban-testnet.stellar.org");
+  const account = await server.getAccount(sourceKeypair.publicKey());
+  const fee = "200100";
+
+  // Create the contract code identifier
+  const contractHash = new StellarSdk.xdr.LedgerKey.contractCode({
+    hash: StellarSdk.Address.fromString(contractId).toScAddress(),
+  });
+
+  // Create footprint for contract code
+  const footprint = new StellarSdk.xdr.LedgerFootprint({
+    readOnly: [],
+    readWrite: [contractHash],
+  });
+
+  // build and submit the transaction
+  const transaction = new StellarSdk.TransactionBuilder(account, {
+    fee,
+    networkPassphrase: StellarSdk.Networks.TESTNET,
+  })
+    .addOperation(
+      StellarSdk.Operation.extendFootprintTtl({
+        extendTo: 500_000,
+        footprint: footprint,
+      }),
+    )
+    .setTimeout(30)
+    .build();
```

Comment:
We probably want to `prepareTransaction` first before it gets sent so that `simulateTransaction` and `assembleTransaction` gets run